chill-scala/src/test/scala-2.13+/com/twitter/chill/SerializedExamplesData.scala

```diff
@@ -148,59 +148,61 @@ object SerializedExamplesData {
-    140 -> ("jgEBCg==" -> new VolatileByteRef(10)),
-    141 -> ("jwEBAQBqYXZhLm1hdGguQmlnRGVjaW1h7AECAgA=" -> math.BigDecimal(2)),
-    142 -> ("kAEBAA==" -> (Queue.empty[Any], true)),
-    143 -> ("kQEBAQIC" -> (Map(1 -> 2).keySet, true))
+    94 -> ("YAEA" -> Vector.empty[Int]),
+    95 -> ("YQEBAgI=" -> Vector(1)),
+    96 -> ("YgECgA==" -> Array(Byte.MinValue)),
+    97 -> ("YwECf/8=" -> Array(Short.MaxValue)),
+    98 -> ("ZAEC/////w8=" -> Array(Int.MinValue)),
+    99 -> ("ZQEC/v//////////" -> Array(Long.MaxValue)),
+    100 -> ("ZgECAAAAAQ==" -> Array(Float.MinPositiveValue)),
+    101 -> ("ZwEC/+////////8=" -> Array(Double.MinValue)),
+    102 -> ("aAECAQ==" -> Array(true)),
+    103 -> ("aQECAHg=" -> Array('x')),
+    104 -> ("agECAWNh9A==" -> Array("cat")),
+    105 -> ("awEDAgQDAW1vdXPl" -> Array(2, "mouse")),
+    106 -> ("bAECAQ==" -> classOf[Int]),
+    107 -> ("bQE=" -> new Object()),
+    108 -> ("bgEBBgFiAQKA" -> wrapByteArray(Array(Byte.MinValue))),
+    109 -> ("bwEBCAFjAQJ//w==" -> wrapShortArray(Array(Short.MaxValue))),
+    110 -> ("cAEBAgFkAQL/////Dw==" -> wrapIntArray(Array(Int.MinValue))),
+    111 -> ("cQEBCQFlAQL+//////////8=" -> wrapLongArray(Array(Long.MaxValue))),
+    112 -> ("cgEBBAFmAQIAAAAB" -> wrapFloatArray(Array(Float.MinPositiveValue))),
+    113 -> ("cwEBCgFnAQL/7////////w==" -> wrapDoubleArray(Array(Double.MinValue))),
+    114 -> ("dAEBBQFoAQIB" -> wrapBooleanArray(Array(true))),
+    115 -> ("dQEBBwFpAQIAeA==" -> wrapCharArray(Array('x'))),
+    116 -> ("dgEBAwBqAQIBY2H0" -> wrapRefArray(Array("cat"))),
+    117 -> ("dwE=" -> None),
+    118 -> ("eAEA" -> collection.immutable.Queue()),
+    119 -> ("eQEA" -> Nil),
+    120 -> ("egEBAgg=" -> (4 :: Nil)),
+    // 121-> ("eAEGAAQEAgI=" -> collection.immutable.Range(1, 3)),
+    122 -> ("fAEBdGHj" -> wrapString("tac")),
+    123 -> ("fQECgQEBAgQCBg==" -> collection.immutable.TreeSet(3, 2)),
+    124 -> ("fgEBgQEBJwECBgIE" -> collection.immutable.TreeMap(3 -> 2)),
+    125 -> ("fwE=" -> math.Ordering.Byte),
+    126 -> ("gAEB" -> math.Ordering.Short),
+    127 -> ("gQEB" -> math.Ordering.Int),
+    128 -> ("ggEB" -> math.Ordering.Long),
+    129 -> ("gwEB" -> math.Ordering.Float),
+    130 -> ("hAEB" -> math.Ordering.Double),
+    131 -> ("hQEB" -> math.Ordering.Boolean),
+    132 -> ("hgEB" -> math.Ordering.Char),
+    133 -> ("hwEB" -> math.Ordering.String),
+    134 -> ("iAEBAA==" -> Set[Any]()),
+    135 -> ("iQEBAA==" -> ListSet[Any]()),
+    136 -> ("igEBAUgBgmE=" -> ListSet[Any]('a)),
+    137 -> ("iwEBAA==" -> Map[Any, Any]()),
+    138 -> ("jAEBAA==" -> ListMap[Any, Any]()),
+    139 -> ("jQEBAScBSAGCYUgE" -> ListMap('a -> 'a)),
+    140 -> ("jgEBegEBAgI=" -> Stream(1)),
+    141 -> ("jwEB" -> Stream()),
+    142 -> ("kAEBCg==" -> new VolatileByteRef(10)),
+    143 -> ("kQEBAQBqYXZhLm1hdGguQmlnRGVjaW1h7AECAgA=" -> math.BigDecimal(2)),
+    144 -> ("kgEBAA==" -> (Queue.empty[Any], true)),
+    145 -> ("kwEBAQIC" -> (Map(1 -> 2).keySet, true))
```

Comment:
I'm not sure the benefit of these tests if we just change them each time. I think this breaks anyone with stored serialized data. I think storing kryo data long term is a really *REALLY* bad idea, but I think we should have a consistent story here (and maybe bump a major version or something if we really change to "all bets are off!").

Reply:
Hi @johnynek, yeah I was thinking the same. After applying this changes I realised that we probably should look into how we actually register classes in kryo since it affects serialization (Kryo registration ids are kept). Maybe we can come up with way to incrementally update. If we do break it then yeah coming up a new major version should be the way to go.

I'm with you as well, storing kryo data should be avoided. The only use case where i think it's ok is taking snapshots for recovery, while making sure versions stay the same.

Reply:
Not sure if we wanna merge this now and tackle what i mentioned afterwards ...

Reply:
> I realised that we probably should look into how we actually register classes in kryo since it affects serialization (Kryo registration ids are kept). 

on a second thought I'll need to spend more time here ... because it might not be as I'm thinking ... in this PR we see that some lower Id's got affected while 94 and 95 were added.

Reply:
> Not sure if we wanna merge this now and tackle what i mentioned afterwards ...

let's wait ... let me see if i come up with smth

Reply:
@johnynek should be good now!